util.go

```diff
@@ -154,11 +155,43 @@ func trimeSchemaRefLinkPrefix(ref string) string {
 	return strings.TrimPrefix(ref, "#/components/schemas/")
 }
 
-func genSchemeaObjectID(pkgName, typeName string) string {
-	typeNameParts := strings.Split(typeName, ".")
-	pkgName = replaceBackslash(pkgName)
+func genSchemaObjectID(pkgName, typeName string, aliases map[string]*string) string {
+	validatedPkgName := checkAndMutatePackageName(pkgName, aliases)
+	validatedTypeName := checkAndMutateTypeName(typeName, aliases)
+	typeNameParts := strings.Split(validatedTypeName, ".")
+	pkgName = replaceBackslash(validatedPkgName)
 	pkgNameParts := strings.Split(pkgName, "/")
```

Comment:
Should this be using `validatedPkgName` rather than `pkgName`? (then we wouldn't need the previous line either, I think)